src/main/kotlin/no/nav/familie/ef/sak/vedtak/historikk/VedtakHistorikkService.kt

```diff
@@ -18,19 +28,30 @@ import java.util.UUID
 @Service
 class VedtakHistorikkService(
     private val fagsakService: FagsakService,
-    private val andelsHistorikkService: AndelsHistorikkService
+    private val andelsHistorikkService: AndelsHistorikkService,
+    private val barnService: BarnService,
+    private val featureToggleService: FeatureToggleService
 ) {
 
+    fun hentVedtakFraDato(behandlingId: UUID, fra: YearMonth): VedtakDto {
+        val fagsak = fagsakService.hentFagsakForBehandling(behandlingId)
+        return when (fagsak.stønadstype) {
+            StønadType.OVERGANGSSTØNAD -> hentVedtakForOvergangsstønadFraDato(fagsak, fra)
+            StønadType.BARNETILSYN -> hentVedtakForBarnetilsynFraDato(fagsak, behandlingId, fra)
+            StønadType.SKOLEPENGER -> error("Støtter ikke henting av skolepenger ")
+        }
+    }
+
+    fun hentVedtakForOvergangsstønadFraDato(fagsakId: UUID, fra: YearMonth): InnvilgelseOvergangsstønad {
+        return hentVedtakForOvergangsstønadFraDato(fagsakService.hentFagsak(fagsakId), fra)
+    }
+
     /**
      * Slår sammen perioder som er sammenhengende, med samme aktivitet, og samme periodetype, unntatt sanksjoner
      * Slår sammen inntekter som er sammenhengende, med samme inntekt og samordningsfradrag
      */
-    fun hentVedtakForOvergangsstønadFraDato(fagsakId: UUID, fra: YearMonth): InnvilgelseOvergangsstønad {
-        val stønadstype = fagsakService.hentFagsak(fagsakId).stønadstype
-        feilHvis(stønadstype != StønadType.OVERGANGSSTØNAD) {
-            "Kan kun hente data for overgangsstønad fra denne, men prøvde å hente for $stønadstype fagsak=$fagsakId"
-        }
-        val historikk = hentAktivHistorikk(fagsakId)
+    private fun hentVedtakForOvergangsstønadFraDato(fagsak: Fagsak, fra: YearMonth): InnvilgelseOvergangsstønad {
+        val historikk = hentAktivHistorikk(fagsak, StønadType.OVERGANGSSTØNAD)
```

Comment:
Tok du bort validering her for evt kunne bruke denne til flere stønadstyper, eller er det andre grunner til at valideringen ble tatt bort her?

Reply:
Ah - hentAktivHistorikk(fagsak, StønadType.OVERGANGSSTØNAD)